rca/utils/blocks.py

```diff
@@ -103,6 +104,46 @@ class GalleryBlock(blocks.StructBlock):
     author = blocks.CharBlock(required=False)
     link = blocks.URLBlock(required=False)
     course = blocks.CharBlock(required=False)
+    document = DocumentChooserBlock(
+        required=False, help_text="Recommended maximum file size: 10MB"
+    )
+    video_embed = EmbedBlock(
+        help_text="Add a YouTube or Vimeo video URL", required=False
+    )
+    audio_embed = EmbedBlock(help_text="Add a Soundcloud URL", required=False)
+
+    def clean(self, value):
+        if bool(value.get("document")):
+            if value.get("document").file_size > 10000000:
+                raise StructBlockValidationError(
+                    {"document": ErrorList(["Please ensure your file is below 10MB"])},
+                )
+
+        if bool(value.get("document")) and bool(value.get("video_embed")):
+            raise StructBlockValidationError(
+                {
+                    "document": ErrorList(
+                        ["Multiple values are not supported for Document and Video."]
+                    )
+                },
+            )
+        if bool(value.get("document")) and bool(value.get("audio_embed")):
+            raise StructBlockValidationError(
+                {
+                    "document": ErrorList(
+                        ["Multiple values are not supported for Document and Audio."]
+                    )
+                },
+            )
+        if bool(value.get("video_embed")) and bool(value.get("audio_embed")):
+            raise StructBlockValidationError(
+                {
+                    "video_embed": ErrorList(
+                        ["Multiple values are not supported for Video and Audio."]
+                    )
+                },
+            )
```

Comment:
Yea great point, thanks

Reply:
I think you'll also need to add checks if `document` is already a key in `errors`

With the current change, if a user uploads a file more than 10MB and also enters a video embed, only `"Multiple values are not supported for Document and Video."` will show since lines 121-123 overwrite the message that was added in lines 116-118